### PR TITLE
mac-capture: Improve window capture CPU usage

### PR DESCRIPTION
**Pull request by jpark37**
### Description
Use data provider instead of color conversion path. Reduces capture loop from
around 125 milliseconds to about 75 millisseconds.

Also sleep capture thread every loop for a rough 50% throttle.

### Motivation and Context
Mac window capture thread is very expensive and bogging people down.

### How Has This Been Tested?
Window capture still appears to work without noticeable regression, CPU percentages are down.

TODO: Which applications are going to capture improperly now that the color conversion has been removed? We may need GPU format conversion shaders, or restore the previous path conditionally.

### Types of changes
- Performance enhancement (non-breaking change which improves efficiency)

### Checklist:
- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

File: plugins/mac-capture/mac-window-capture.m [hunk 1]
```diff
@@ -50,28 +46,22 @@ static inline void capture_frame(struct window_capture *wc)
...
 		.width = width,
 		.height = height,
-		.data[0] = data,
+		.data[0] = bytes,
 		.linesize[0] = width * 4,
 		.timestamp = ts,
 	};
 
 	obs_source_output_video(wc->source, &frame);
+	CFRelease(data);
```
Comment (ePirat) [inline plugins/mac-capture/mac-window-capture.m:64]: Does `obs_source_output_video` copies the frame `data`? Else this change will introduce a use after free, if data is still used after the underlying CFData is released.

Comment (jpark37) [reply plugins/mac-capture/mac-window-capture.m:64]: It looks to me like it copies the data.

obs_source_output_video -> obs_source_output_video_internal -> cache_video -> copy_frame_data

Comment (ePirat) [reply plugins/mac-capture/mac-window-capture.m:64]: Indeed, looks like it. Then this is fine.

**Comment by ePirat** [comment]
So overall this looks good, my biggest concern is the lack of colorspace conversion thats introduced by this change…

Another useful improvement would be to get rid of the sleeping and find some way of knowing when the contents actually changed… Although not sure there is public API that would allow for this…

**Comment by jpark37** [comment]
I'm leaning toward killing off this PR because I barely remember what I did, and i don't want to sign up to support Mac regressions. Maybe Qt 5.15.1 by itself will make Mac users happy.

**Comment by jpark37** [comment]
Maybe I'll revisit this another day, or someone else can drive home something like this.

**Pull request closed without merging**